Source/src/Batching/BatchManager.cpp

```diff
@@ -192,9 +192,13 @@ void Hachiko::BatchManager::ShowDebugMenuForBatches(
 void Hachiko::BatchManager::DrawSingleBatch(GeometryBatch* geometry_batch, const Program* program, int segment) const 
 {
     // Update and bind batch
-    geometry_batch->UpdateCommands();
-    geometry_batch->UpdateBatch(segment);
-    geometry_batch->BindBatch(segment, program);
+    if (geometry_batch->dirty_draw_components)
```

Comment:
Why did you add this check?

Reply:
Because i call this method two times in a row and it was a waste of resources to repeat the process